static/js/components/nl_search_bar/auto_complete_input.tsx

```diff
@@ -131,18 +131,30 @@ export function AutoCompleteInput(
   const { placeholder } = useQueryStore();
 
   useEffect(() => {
-    window.addEventListener("scroll", () => {
+    const handleScroll = () => {
       setLastScrollY(window.scrollY);
-    });
+    };
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [setLastScrollY]);
 
+  useEffect(() => {
     if (!inputText && props.enableDynamicPlaceholders) {
       enableDynamicPlacehoder(
         setSampleQuestionText,
         setDynamicPlaceholdersEnabled,
         setDynamicPlaceholdersDone
       );
     }
-  }, []);
+  }, [
+    inputText,
+    props.enableDynamicPlaceholders,
+    setSampleQuestionText,
```

Comment:
I didn't know that the state setter methods would/could count as dependencies for useEffect, interesting!